File: 2015/day5.py
import re

test = ['ugknbfddgicrmopn']

nice = 0

for line in open("day5_input.txt").readlines():

  if 'ab' in line or 'cd' in line or 'pq' in line or 'xy' in line:
    continue

  for char in line:
    if line.count(char + char) > 0:
      if line.count('a')+line.count('e')+line.count('i')+line.count('o')+line.count('u') >= 3:
        nice += 1
        break
    
    else:
      continue

print("Part 1:", nice)

# Part 2

test = ['qjhvhtzxzqqjkmpb']
nice = 0

for line in open("day5_input.txt").readlines():
  if re.search(rf"([a-z]).\1", line) and re.search(rf"([a-z][a-z]).*\1", line):
    nice += 1

print("Part 2:", nice)

File: 2015/Day5.rb
# keywords: =~, pattern matching

# Day 5
# Match for 3 vowels, at least 1 letter twice in a row, and
# does not contain ab, cd, pq, or xy

# test = 'ugknbfddgicrmopn'
# c = 0

data = File.readlines('Day5_input.txt')

# data.each do |line|
#     puts line
#     badMatch = false

#     ["ab", "cd", "pq", "xy"].each do | p |
#         if line.include?(p)
#             badMatch = true
#             break
#         end
#     end

#     if (/(.)\1/ =~ line) && !badMatch
#         if (line.count "aeiou") >= 3
#             c += 1
#         end
#     end
# end

# puts c

# PART 2

test = 'qjhvhtzxzqqjkmpb'
c = 0

data.each do |line|
  # puts line
  if line =~ /([a-z][a-z]).*\1/
    puts 'Pairs found'
    if line =~ /([a-z]).\1/
      puts "#{line.chomp} is nice"
      c += 1
    else
      puts "#{line.chomp} is naughty 2"
    end
  else
    puts "#{line.chomp} is naughty 1"
  end
end
puts c